### DISCUSSION TITLE
next.13 에서는 layout에서는 styled-component 쓰는게 불가능 한가

### DISCUSSION THREAD

**Discussion by sumi-0011** [probrainteam/GoaTani-client#4] [General]
## bug code
import styled from 'styled-components'
import './globals.css'

export const metadata = {
    title: 'GoaTani',
    description: 'G.O.A.T Ohtani',
}

export default function RootLayout({
    children,
}: {
    children: React.ReactNode
}) {
    return (
        <html lang="ko">
            <body>{children}</body>
        </html>
    )
}

const Wrapper = styled.div`
    height: 59px;
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 0px 18px;
`

## 스크린샷
![image](https://user-images.githubusercontent.com/49177223/229794274-b8c488b6-bbb0-4ede-b3f4-2e4bc34e1312.png)

**Comment by sumi-0011** [comment]
## 해결 방법
layout.tsx 파일의 최상단에 'use client'를 추가하여 해결

## 추가 버그
- The Next.js 'metadata' API is not allowed in a client component.ts(71008)
- client 컴포넌트에서는 서버 로직인 metadata를 허용하지 않아서 발생, 제거함으로써 문제 해결
```
export const metadata = {
    title: 'GoaTani',
    description: 'G.O.A.T Ohtani',
}
```
![image](https://user-images.githubusercontent.com/49177223/229794342-0c06bd8c-fd60-4457-8749-135690ba8c6d.png)

**Comment by sumi-0011** [comment]
Next.js 13에서 root layout은 _app, _document를 대체하는거라서 styled-component 자체를 사용하면 안될것 같다. 
![image](https://user-images.githubusercontent.com/49177223/229794821-dd97f689-87ed-4021-9fd1-9898c86edf7b.png)

**Reply by sumi-0011** [reply]
근데 그러면 `styled-component`의 `ThemeProvider`은 어디에 감싸야 하는걸까요?

**Reply by sumi-0011** [reply]
### 참고 
https://velog.io/@qqww08/Next-13-%EC%97%90%EC%84%9C%EC%9D%98-CSS-in-JS-with.-styled-components

**Comment by jeonbyeongmin** [comment]
root layout 은 _app, _document 파일을 대체하도록 되어 있어요.
만약 root layout 에 provider를 감싸고 싶을 때에는 해당 provider를 분리해서 그 파일에는 "use client"를 붙여주면 되는 것 같아요.
그리고 그 provider를 root layout에서 import하고 사용하면 될 것 같습니다!

아래 링크도 참고해보아요
https://github.com/styled-components/styled-components/issues/3902
